### PR TITLE
test: disable relay and enable relay according to version

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 

- upgrade and downgrade CI with version v2.0.1 fails
- didn't test relay in versions later than 2.0.2

### What is changed and how it works?

as title

### Check List 

Tests 

 - No code

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: tests/tiup/upgrade-from-v2.sh [hunk 1]
```diff
@@ -27,10 +27,15 @@ function deploy_previous_v2() {
 function migrate_in_previous_v2() {
 	exec_full_stage
 
-	# v2.0.0 doesn't implement relay log, and enable-gtid for MariaDB has a bug
-	if [[ "$PRE_VER" == "v2.0.0" ]]; then
-		sed -i "s/enable-relay: true/enable-relay: false/g" $CUR/conf/source2.yaml
-		sed -i "s/enable-gtid: true/enable-gtid: false/g" $CUR/conf/source2.yaml
+	# v2.0.0 doesn't implement relay log.
+	# v2.0.1 downgrading has a problem, we use config import which create sources and then create tasks, so for a cluster
```
Comment (lance6716) [inline tests/tiup/upgrade-from-v2.sh:31]: maybe we can let `config export` get location of `getMinLocInAllSubTasks` and write it to source config's relay location? @GMHDBJD

File: tests/tiup/lib.sh [hunk 2]
```diff
@@ -261,3 +261,14 @@ function run_dmctl_with_retry() {
...
 }
+
+function ensure_start_relay() {
+	# manually enable relay for source1 after v2.0.2
+	if [[ "$PRE_VER" != "v2.0.0" ]] && [[ "$PRE_VER" != "v2.0.1" ]]; then
+		dmctl_log="get-worker.txt"
+		# always use CUR_VER, because we might use tiup mirror in previous steps.
+		tiup dmctl:$CUR_VER --master-addr=master1:8261 operate-source show -s mysql-replica-01 >$dmctl_log 2>&1
+		worker=$(grep "worker" $dmctl_log | awk -F'"' '{ print $4 }')
+		run_dmctl_with_retry $CUR_VER "start-relay -s mysql-replica-01 $worker" "\"result\": true" 1
```
Comment (GMHDBJD) [inline tests/tiup/lib.sh:272]: Not use retry?

**Review by GMHDBJD — COMMENTED** [review]
rest LGTM

File: tests/tiup/lib.sh (hunk 2 above)
Comment (lance6716) [reply tests/tiup/lib.sh:272]: the first time should succeed. I just want to reuse existing function to do a result checking.

**Review by GMHDBJD — APPROVED** [review]
/LGTM

**Review by Ehco1996 — APPROVED** [review]

**Pull request merged**